esp32_s3_korvo_2/esp32_s3_korvo_2.c

```diff
@@ -0,0 +1,596 @@
+            .swap_xy = 0,
+            .mirror_x = 1,
+            .mirror_y = 0,
+        },
+    };
+    esp_lcd_panel_io_handle_t tp_io_handle = NULL;
+    const esp_lcd_panel_io_i2c_config_t tp_io_config = ESP_LCD_TOUCH_IO_I2C_TT21100_CONFIG();
+    BSP_ERROR_CHECK_RETURN_ERR(esp_lcd_new_panel_io_i2c((esp_lcd_i2c_bus_handle_t)BSP_I2C_NUM, &tp_io_config, &tp_io_handle));
+    BSP_ERROR_CHECK_RETURN_ERR(esp_lcd_touch_new_i2c_tt21100(tp_io_handle, &tp_cfg, &tp));
+    assert(tp);
+
+    /* Register a touchpad input device */
+    lv_indev_drv_init(&indev_drv_tp);
+    indev_drv_tp.type = LV_INDEV_TYPE_POINTER;
+    indev_drv_tp.read_cb = bsp_touchpad_read;
+    indev_drv_tp.user_data = tp;
+    indev_touchpad = lv_indev_drv_register(&indev_drv_tp);
+    BSP_NULL_CHECK(indev_touchpad, ESP_ERR_NO_MEM);
+
+    return ESP_OK;
+}
+#endif //BSP_DISPLAY_ENABLED
+
+esp_err_t bsp_display_brightness_set(int brightness_percent)
+{
+    ESP_LOGW(TAG, "This board doesn't support to change brightness of LCD");
+    return ESP_ERR_NOT_SUPPORTED;
+}
+
+esp_err_t bsp_display_backlight_off(void)
+{
+    return bsp_display_brightness_set(0);
+}
+
+esp_err_t bsp_display_backlight_on(void)
+{
+    return bsp_display_brightness_set(100);
+}
+
+lv_disp_t *bsp_display_start(void)
+{
+#if BSP_DISPLAY_ENABLED
+    lv_init();
+    lv_disp_t *disp = lvgl_port_display_init();
+    BSP_ERROR_CHECK_RETURN_NULL(lvgl_port_indev_init());
+    BSP_ERROR_CHECK_RETURN_NULL(lvgl_port_tick_init());
+    lvgl_mux = xSemaphoreCreateMutex();
+    BSP_NULL_CHECK(lvgl_mux, NULL);
+    xTaskCreate(lvgl_port_task, "LVGL task", 4096, NULL, CONFIG_BSP_DISPLAY_LVGL_TASK_PRIORITY, NULL);
+    return disp;
+#else
+    ESP_LOGW(TAG, "Display is disabled, because it is not tested yet!");
+    return NULL;
+#endif
+}
+
+void bsp_display_rotate(lv_disp_t *disp, lv_disp_rot_t rotation)
+{
+#if BSP_DISPLAY_ENABLED
+    lv_disp_set_rotation(disp, rotation);
+#else
+    ESP_LOGW(TAG, "Display is disabled, because it is not tested yet!");
+#endif
+}
+
+bool bsp_display_lock(uint32_t timeout_ms)
+{
+#if BSP_DISPLAY_ENABLED
+    assert(lvgl_mux && "bsp_display_start must be called first");
+
+    const TickType_t timeout_ticks = (timeout_ms == 0) ? portMAX_DELAY : pdMS_TO_TICKS(timeout_ms);
+    return xSemaphoreTake(lvgl_mux, timeout_ticks) == pdTRUE;
+#else
+    return pdFALSE;
+#endif
+}
+
+void bsp_display_unlock(void)
+{
+#if BSP_DISPLAY_ENABLED
+    assert(lvgl_mux && "bsp_display_start must be called first");
+    xSemaphoreGive(lvgl_mux);
+#endif
+}
+
+esp_err_t bsp_button_init(const bsp_button_t btn)
+{
+    const gpio_config_t button_io_config = {
+        .pin_bit_mask = BIT64(btn),
+        .mode = GPIO_MODE_INPUT,
+        .pull_up_en = GPIO_PULLUP_DISABLE,
+        .pull_down_en = GPIO_PULLDOWN_DISABLE,
+        .intr_type = GPIO_INTR_DISABLE
+    };
+    return gpio_config(&button_io_config);
+}
+
+bool bsp_button_get(const bsp_button_t btn)
+{
+    return !(bool)gpio_get_level(btn);
+}
```

Comment:
Is this even working? Have a look at the Korvo schematic: https://dl.espressif.com/dl/schematics/SCH_ESP32-S3-KORVO-2_V3_0_20210918.pdf

All buttons share one GPIO. You must measure voltage on the pin. These two functions should use `iot_button` API

Reply:
I know, it is in example. I forgot remove this part of code. Thanks for note.